android/src/com/metalrain/stocksimulator/android/views/RandomEventView.java

```diff
@@ -0,0 +1,53 @@
+package com.metalrain.stocksimulator.android.views;
+
+import android.app.Activity;
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.TextView;
+
+import com.metalrain.stocksimulator.android.StockSimulator;
+import com.metalrain.stocksimulator.state.components.StockImpactEventComponent;
+
+import rx.Subscription;
+import rx.functions.Action1;
+
+/**
+ * Created by octopoco on 7/26/2015.
+ */
+public class RandomEventView extends TextView {
+Context context;
+    private Subscription subscription;
+    public RandomEventView(Context context){
+        super(context);
+        this.context=context;
+        setText("test");
+    }
+    public RandomEventView(Context context, AttributeSet attributeSet){
+        super(context,attributeSet);
+        this.context=context;
+        setText("test");
+    }
+    @Override
+    public void onAttachedToWindow(){
+        super.onAttachedToWindow();
+        StockSimulator.getGameState().startThread();
```

Comment:
This might be a bug, starting/pausing the game isn't handled here. It's in the activity lifecycle.